### PR TITLE
T250800 Setup Translatewiki

### PR DESCRIPTION
**Pull request by hueitan**
Phabricator Ticket : https://phabricator.wikimedia.org/T250800

 - [x] Adding Translatewiki needed folder and files
 - [x] Scripting our own i18n utils, and make it flexible enough to migrate banana when needed
    - [x] Feature needed : Proper language fallback
    - [x] Feature needed : Return the key when no found
 - [x] <s>Setup webpack to include json files as an object</s> webpack includes `json-loader` by default

### CODE REVIEW

File: src/i18n.js [hunk 1]
```diff
@@ -1,17 +1,14 @@
...
-		'wikipedia': 'Wikipédia'
-	},
-	es: {
-		'wikipedia': 'Wikipedia'
+import fallbackMessage from '../i18n/en.json'
+const msg = ( lang, key ) => {
+	let messages
+
+	try {
+		messages = require( `../i18n/${lang}.json` )
```
Comment (stephanebisson) [inline src/i18n.js:6]: Is this going to load the file every time the `msg` function is called?

Comment (hueitan) [reply src/i18n.js:6]: nope, the file (object) is being stored within the internal code.

Comment (hueitan) [reply src/i18n.js:6]: sorry, got the wrong understanding here, true, we should make it available once the require is called for the next call used.

Comment (hueitan) [reply src/i18n.js]: done, and please check it again.

File: src/i18n.js [hunk 2]
```diff
@@ -1,17 +1,20 @@
...
+	msg = ( lang, key ) => {
+		try {
+			if ( !messages[ lang ] ) {
+				messages[ lang ] = require( `../i18n/${lang}.json` )
+			}
+		} catch ( error ) {
+			// Translation not available
+		}
+
+		let message = messages[ lang ]
```
Comment (stephanebisson) [inline src/i18n.js:16]: can be `const`

File: src/i18n.js [hunk 3]
```diff
@@ -1,17 +1,20 @@
...
-		'wikipedia': 'Wikipedia'
-	}
-}
 
-const msg = (lang, key) => {
-	return (messages[lang] || messages.en)[key] || key
-}
+	msg = ( lang, key ) => {
+		try {
+			if ( !messages[ lang ] ) {
```
Comment (stephanebisson) [inline src/i18n.js:9]: I would suggest reworking this logic to not retry to load a missing language all the time. We could put the catch around the `require` and maybe initialize the messages with an empty object when it fails.

Comment (hueitan) [reply src/i18n.js]: update, thanks it looks better now.

**Pull request merged**